resources/systemd_timer.rb

```diff
@@ -0,0 +1,123 @@
+#
+# Cookbook:: chef-client
+# resource:: chef_client_systemd_timer
+#
+# Copyright:: 2020, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :chef_client_systemd_timer
+
+property :user, String, default: 'root'
+
+property :job_name, String, default: 'chef-client'
+property :delay_after_boot, String, default: '1min'
+property :interval, String, default: '30min'
+
+property :splay, [Integer, String], default: 300,
+                                    coerce: proc { |x| Integer(x) },
+                                    callbacks: { 'should be a positive number' => proc { |v| v > 0 } }
+
+property :description, String, default: 'Chef Infra Client periodic execution'
```

Comment:
Should this use the `Chef::Dist` values for the product name and log directory and all that?

Reply:
We're supporting chef releases before the Dist stuff was added in this cookbook.